api/model/api.go

```diff
@@ -983,6 +982,14 @@ type GraphQLConfig struct {
 
 	// GraphQLPlayground is the Playground specific configuration.
 	GraphQLPlayground GraphQLPlayground `json:"playground,omitempty"`
+
+	// Proxy holds the configuration for a proxy only api.
+	Proxy GraphQLProxyConfig `bson:"proxy" json:"proxy,omitempty"`
+}
+
+type GraphQLProxyConfig struct {
+	// +nullable
+	AuthHeaders map[string]string `bson:"auth_headers" json:"auth_headers,omitempty"`
```

Comment:
Once users use `proxy` field within their `GraphQLConfig`, shouldn't they need to provide `AuthHeaders`? If not, is it okay to provide `AuthHeaders` as null when `Proxy` is declared?

Also, do we need to use `bson` struct tag?

Reply:
You are right. AuthHeaders should not be nullable. 
I will fix that and also remove bson tag.

Reply:
What is different about this field in comparison with a more generic option such as UpstreamHeaders? e.g. does it automatically inject "Bearer " prefix? Why would you need multiple auth headers?

It would be good to document this difference (if any) so that the docs propagate to the CRD. Maybe i'm looking too much into it, but im not sure why it would be named as-such.

Reply:
![Screenshot from 2022-03-23 17-27-18](https://user-images.githubusercontent.com/5239161/159699881-46ae0322-382e-4f9e-818d-6800899fb881.png)
@asoorm Are you referring to `Upstream Auth Headers` field on this screen?
If you are, it sets `proxy.auth_headers` field in the APIDefinition

Reply:
@buraksekili If I remove nullable for AuthHeader, API Definition creation fails if AuthHeaders aren't specified with error stating "Object can't have null value". Have to add it back